src/icloudpd/authentication.py

```diff
@@ -29,8 +29,7 @@ def authenticate_(
                 # If password not provided on command line variable will be set to None
                 # and PyiCloud will attempt to retrieve from its keyring
                 icloud = pyicloud_ipd.PyiCloudService(
-                    domain,
```

Comment:
Will need to address this removal.

Reply:
> Will need to address this removal.

domain parameter did not work as expected for China. I suspect that there is more in auth than just domain change.

As a mitigation approach, we can make icloudpd working without domain first and then see how China can be addressed. Many ppl are using icloudpd from China, so it is important, but developing/troubleshooting without access from China itself is hard if possible at all. I was not able to reproduce access issues from China using VPN (tunneling from US to China servers)...

Reply:
> As a mitigation approach, we can make icloudpd working without domain first

Ok, that's largely what I've done in a naive way since I'm not truly familiar with all the pieces at play here. My primary goal was to see if the upstream changes to icloudpd would get me further along in the auth process, and so far, so good. I've not tried any testing related to China domains, etc.

At the moment, I'm stuck trying to get a docker container working, and I may have missed some setup instructions.

- I copied Dockerfile to Dockerfile.release (since that was required as part of `scripts/build_docker`
- Then ran `scripts/build_docker`

After that, I tried running this:

```sh
docker run -it \
--rm --name icloudpd \
-v ~/icloudpd-download:/data \
-e TZ=America/New_York \
icloudpd/icloudpd:latest \
  icloudpd \
  --directory /data \
  --username my-email@example.com \
  --watch-with-interval 3600
```

But hit a snag, and have more code to update:

```sh
Traceback (most recent call last):
  File "starters/icloudpd_ex.py", line 7, in <module>
  File "PyInstaller/loader/pyimod02_importers.py", line 385, in exec_module
  File "icloudpd/base.py", line 20, in <module>
  File "PyInstaller/loader/pyimod02_importers.py", line 385, in exec_module
  File "pyicloud_ipd/__init__.py", line 2, in <module>
  File "PyInstaller/loader/pyimod02_importers.py", line 385, in exec_module
  File "pyicloud_ipd/base.py", line 12, in <module>
ImportError: cannot import name 'PyiCloudConnectionException' from 'pyicloud_ipd.exceptions' (/tmp/_MEIlfLIgg/pyicloud_ipd/exceptions.pyc)
[8] Failed to execute script 'icloudpd_ex' due to unhandled exception!
```

Reply:
> At the moment, I'm stuck trying to get a docker container working, and I may have missed some setup instructions.

Check .github subfolder for how docker is built for release/check. I use `docker build . -t icloudpd_dev` for local testing. I'll check `scripts/build_docker` is still relevant, suspect it is not.